mathlibtools/file_status.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+from typing import Optional, Set
+
+@dataclass(frozen=True)
+class FileStatus:
+  """Capture the file status descriptions in the wiki yaml.
+
+  `string_match` are the substrings that must be found, lowercase match
+  `color` is how the node should be colored if it has the status
+  """
+
+  string_match: Set[str]
+  # colors from X11
+  color: str
+
+  @classmethod
+  def yes(cls) -> "FileStatus":
+    return cls({"yes"}, "green")
+
+  @classmethod
+  def pr(cls) -> "FileStatus":
+    return cls({"no", "pr"}, "lightskyblue")
+
+  @classmethod
+  def wip(cls) -> "FileStatus":
+    return cls({"no", "wip"}, "lightpink")
+
+  @classmethod
+  def no(cls) -> "FileStatus":
+    return cls({"no"}, "orange")
+
+  @classmethod
+  def missing(cls) -> "FileStatus":
+    return cls(set(), "orchid1")
+
+  @classmethod
+  def ready(cls) -> "FileStatus":
+    return cls(set(), "turquoise1")
+
+  def matches(self, comment: str) -> bool:
+    return all(substring.lower() in comment.lower() for substring in self.string_match)
```

Comment:
Oh, I don't like this. We care about the `yes` and `no` as initial strings, not anything afterwards.